grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -807,14 +807,35 @@ class StockMovementApiController {
     }
 
     def requisitionStatusCodes = {
-        def options = RequisitionStatus.listOutboundOptions()?.collect {
-            [
-                    id: it.name(),
-                    value: it.name(),
-                    label: "${g.message(code: 'enum.RequisitionStatus.' + it.name())}",
-                    variant: it.variant.name
-            ]
+        boolean hasRequestApprovalActivity = false
+        boolean isRequestsList = params.getBoolean("isRequestsList")
+        if (params.location) {
+            Location location = Location.get(params.location)
+            hasRequestApprovalActivity = location.supports(ActivityCode.APPROVE_REQUEST)
         }
-        render([data: options] as JSON)
+
+        List<RequisitionStatus> statuses = getRequisitionStatusCodesList(hasRequestApprovalActivity, isRequestsList)
+
+        List options = statuses?.collect(mapStatusCodes)
+        List availableStatuses = RequisitionStatus.listAll().collect(mapStatusCodes)
```

Comment:
I had to split options on the dropdown from statuses that we can have on the list page. In the frontend part, we used fetched statuses from the dropdown to display their names on the list, but when we removed the verified status we still had at least one stock movement with this status, and the error was thrown. So I split those two cases (We were trying to show the status that we didn't fetch). Regarding the validation with visible requests in "approved"/"waiting for approval" status - we have a ticket for this and I think this should be improved/resolved here

Reply:
if `options` are still based on what `getRequisitionStatusCodesList` returns, can't we still run into this problem, since this method can filter out some of the statuses?

Reply:
I think I understand the problem but not fully. To me, this approach doesn't seem right but maybe I am missing some context.
Anyway, we will probably need to discuss this part further.